modules/loggers/mod_log_config.c

```diff
@@ -1513,6 +1624,10 @@ static int open_multi_logs(server_rec *s, apr_pool_t *p)
                 format = apr_table_get(mls->formats, cls->format_string);
                 if (format) {
                     cls->format = parse_log_string(p, format, &dummy);
+                    if(cls->format && cls->log_formatter == ap_json_log_formatter /* sort_log_format_items */ ) {
+                        /* sort log_format_items array, to help json formatter */
+                        qsort(cls->format->elts, cls->format->nelts, cls->format->elt_size, compare_log_format_item_by_tag);
```

Comment:
Do we need to do this both here and in add_custom_log()?

Reply:
I did think about this one a bit today and pushed the commit https://github.com/apache/httpd/pull/373/commits/9f8355c35977ab7811afb60555b2e9303517bf80 - what do you think?

Reply:
thinking a bit more about this I think the lookup of log formats happening currently in "ap_hook_open_logs(init_config_log" should IMHO actually be done in "ap_hook_check_config(log_check_config" as part of consistency check of the current configuration.

So for example having a:
CustomLog "logs/access_log" combind

Will get silently parsed as an empty format string, and not rejected as an error of misspelled format nickname "combined", which is probably not what the user wants.

Also the check in "config_log_transaction":
format = cls->format ? cls->format : default_format;
can actually only happen in TransferLog case in which case fallback to default format is done.

Maybe rejecting format string which doesn't contain any format specifier, if it's not a format nickname is a sensible change?

and moving all those checks/consolidating into log_check_config let's probably get rid of some invariants.
but thats probably stuff for another PR...

Reply:
> Maybe rejecting format string which doesn't contain any format specifier, if it's not a format nickname is a sensible change?

+1, I have seen it bite people many times and I don't see a reason to preserve it.